apps/rush-lib/src/data/commandLineConfiguration.schema.json

```diff
@@ -0,0 +1,145 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Rush Configuration",
+  "description": "Configuration file for the Rush multi-package build tool",
+
+  "type": "object",
+  "additionalProperties": false,
+
+  "properties": {
+    "$schema": {
+      "description": "Part of the JSON Schema standard, this optional keyword declares the URL of the schema that the file conforms to. Editors may download the schema and use it to perform syntax highlighting.",
+      "type": "string"
+    },
+
+    "customCommands": {
+      "title": "Custom Commands",
+      "description": "A list of custom commands that can be invoked from the Rush command line",
+      "type": "array",
+      "items": {
+        "title": "Custom Command",
+        "type": "object",
+        "additionalProperties": false,
+        "required": [ "name", "description" ],
+        "properties": {
+          "name": {
+            "title": "Custom Command Name",
+            "description": "The name of the custom command, which can be invoked via \"rush <name>\"",
+            "type": "string"
+          },
+          "description": {
+            "title": "Custom Command Description",
+            "description": "A short description of the custom command, which will appear when printing command line usage (e.g. \"rush --help\")",
+            "type": "string"
+          }
+        }
+      }
+    },
+
+    "customOptions": {
+      "title": "Custom Options",
+      "description": "A list of custom options that can be associated with commands",
+      "type": "object",
+      "additionalProperties": false,
+      "patternProperties": {
+        "^--[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*$": {
+          "title": "Custom Option",
+          "type": "object",
+
+          "required": [
+            "description",
+            "supportedCommands"
+          ],
+
+          "properties": {
+            "description": {
+              "title": "Description",
+              "type": "string"
+            },
+
+            "supportedCommands": {
```

Comment:
>supportedCommands [](start = 13, length = 17)

Looking at your sample file, I think maybe "associatedCommands" would be a more intuitive name, since this is a reverse mapping